src/main.rs

```diff
@@ -120,14 +120,22 @@ fn main() {
     #[cfg(feature = "flame")]
     flame::start("print");
 
-    for i in 0..p.segments.len() {
-        p.segments[i].escape(p.shell);
-        p.segments[i].print(p.segments.get(i+1), p.shell, &p.theme);
+    if matches.is_present("rtl") {
+        let n = p.segments.len();
+        for i in 1..n+1 {
+            p.segments[n-i].escape(p.shell);
+            p.segments[n-i].print_rtl(p.segments.get(n-i+1), p.shell, &p.theme);
+        }
+    } else {
+        for i in 0..p.segments.len() {
```

Comment:
I know this isn't new code, and you have no obligation to change it, but it would be cool if you could be so nice as to change this to

```rust
for &[ref current, ref next] in p.segments.windows(2) {
    // ...
}
```

Because then you could change that

```rust
for i in 1..n+1 {
    // ...
}
```

into

```rust
for &[ref current, ref next] in p.segments.windows(2).rev() {
    // ...
}
```

which would be much more clear, as I'm sure you agree

Reply:
Hello,

I do have a bit of spare time to spend on this at the moment ; I come up with the following loop :

``` rust
for sublist in p.segments.windows(2) {
  if let &[ref current, ref next] = sublist {
    current.escape(p.shell);
    current.print(Some(next), p.shell, &p.theme);
  }
}
```
The `if let` construct seems mandatory, as `.windows()` iterables might return a `None()` option.
But this code fails to build, as `current` is immutable, and `.escape()` does modify it - I think it's the line that says
``` rust
self.escaped = true
```
At the moment I can't find a way to get around this. Using `&[ref mut current, ref next]` fails with E0596 - cannot borrow as mutable, this operator (supposedly `.windows()`) yields `&` references.

Any clue ?

Reply:
Oh, you're right. What I suggested just isn't possible, mainly because of the mutability issue but also because I want that last `next` to be `None`. Sorry about leading you on a false scent, I didn't look closely enough

Reply:
If you revert the latest patch that caused compilation issues I'll merge it for now, until I can figure out a better way to do stuff :)